sfge/src/main/java/com/salesforce/rules/ThreadableRuleExecutor.java

```diff
@@ -182,21 +179,29 @@ public void run() {
                             new Violation.TimeoutViolation(
                                     "Path evaluation timed out after " + TIMEOUT + " ms",
                                     submission.getPathEntry()));
+                    if (LOGGER.isErrorEnabled()) {
+                        LOGGER.error("Timeout Error executing rule. submission=" + submission, ex);
+                    }
                 } else {
                     // If the thread threw another exception, wrap it in a violation, including the
-                    // last line that executed
+                    // last 3 frames that got executed.
+                    String frameString =
+                            Arrays.stream(ex.getStackTrace())
+                                    .limit(3)
```

Comment:
I figure three frames should be enough.

Reply:
@jfeingold35 have you checked to see if the three frames here include useful content outside of ThreadableRuleExecutor? Wonder if we need to get lines 3 to 7 instead.